library/cc/request_headers.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <optional>
+
+#include "headers.h"
+#include "request_headers_builder.h"
+#include "request_method.h"
+#include "retry_policy.h"
+#include "upstream_http_protocol.h"
+
+class RequestHeaders : public Headers {
+public:
+  RequestMethod request_method() const;
+  std::string scheme() const;
```

Comment:
this can probably just be `const std::string&` and avoid copying strings here, same for the rest of them